code/modules/ruins/lavalandruin_code/biodome_winter.dm

```diff
@@ -42,7 +42,9 @@
 			hit_object.visible_message(span_warning("[hit_object] is freeze-proof!"))
 			return
 		if(!(hit_object.obj_flags & FROZEN))
-			hit_object.make_frozen_visual()
+			ADD_TRAIT(hit_object, TRAIT_ON_ICE, type)
+			hit_object.make_frozen() //immediately freeze
```

Comment:
why do you need to add a trait to an object just to call a proc and then remove the trait

Reply:
I don't like it either but the proc doesn't work without the trait. How would you do it?